src/torrent_summary.rs

```diff
@@ -146,4 +167,85 @@ impl TorrentSummary {
 
     table
   }
+
+  pub(crate) fn write_json(&self, env: &mut Env) -> Result<()> {
+    let json_struct = self.json_struct();
+    write!(env.out_mut(), "{}", serde_json::to_string(&json_struct)?).unwrap();
+    Ok(())
+  }
+
+  fn json_struct(&self) -> TorrentSummaryData {
+    let mut data = TorrentSummaryData::default();
```

Comment:
Can this be refactored to build the struct in place, i.e.:

```rust
TorrentSummaryData {
  name: self.metainfo.info.name.to_string();
  comment: self.metainfo.comment.map(str::to_string),
  ...
}
``

I prefer that style slightly more, since it avoids missing fields, and in general I like to avoid mutability. (Not that there's any particularly strong reason to avoid it here.)